### PR TITLE
Added k8s support

### PR DESCRIPTION
**Pull request by B-a-S**
Added support of github actions in K8s cluster

https://github.com/B-a-S/nccl-rdma-sharp-plugins/actions/runs/4790930194

### CODE REVIEW

**Comment by artemry-nv** [comment]
@Al3xR01 please review

File: .ci/ci_functions.sh [hunk 1]
```diff
@@ -0,0 +1,46 @@
+function configure () {
```
Comment (artemry-nv) [inline .ci/ci_functions.sh:1]: Please add some comments what is this function for. Check also other functions in the file

File: .ci/ci_functions.sh [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    rm -rf $NFS_WORKSPACE
```
Comment (artemry-nv) [inline .ci/ci_functions.sh:2]: Please verify all the shell scripts with shellcheck and fix the issues (e.g. it's integrated into PyCharm IDE)

File: .ci/ci_functions.sh [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    rm -rf ./nccl-rdma-sharp-plugins
+    cd $NFS_WORKSPACE_ROOT
+    git clone git@github.com:Mellanox/nccl-rdma-sharp-plugins.git
```
Comment (artemry-nv) [inline .ci/ci_functions.sh:5]: If it's clone just for the cfg files, maybe make it shallow?

File: .ci/ci_functions.sh [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    git clone git@github.com:Mellanox/nccl-rdma-sharp-plugins.git 
+
+    cp -r ./SRC/.ci ./nccl-rdma-sharp-plugins/
+    rm -rf ./nccl-rdma-sharp-plugins/.ci/cfg/*
+
+    mkdir -p ./nccl-rdma-sharp-plugins/.ci/cfg/$HOST1/sharp_conf
+    mkdir -p ./nccl-rdma-sharp-plugins/.ci/cfg/$HOST2/sharp_conf
+
+    printf "$HOST1\n$HOST2\n">./nccl-rdma-sharp-plugins/.ci/cfg/$HOST1/hostfile
+    printf "log_verbosity 5\n">./nccl-rdma-sharp-plugins/.ci/cfg/$HOST1/sharp_conf/sharp_am.cfg
```
Comment (artemry-nv) [inline .ci/ci_functions.sh:14]: Do we really need log level 5? Maybe 3 is enough?

File: .ci/ci_functions.sh [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    ( echo "Build FAILED" && tail /GIT/build_nccl_rdma_sharp_plugins.log )
+}
+
+function sharp () {
+    echo "Running configure_sharp.sh..."
+
+    ${WORKSPACE}/.ci/configure_sharp.sh && echo "Step configure_sharp SUCCESFULL !!!" 
+}
+
+function test () {
```
Comment (artemry-nv) [inline .ci/ci_functions.sh:38]: Fix code formatting (e.g. use auto code-formatting from PyCharm or other IDE)

File: .ci/ci_functions.sh [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+git clone https://github.com/NVIDIA/nccl-tests.git $NFS_WORKSPACE/nccl-tests
```
Comment (artemry-nv) [inline .ci/ci_functions.sh:40]: Maybe change to shallow clone?

File: .ci/build_cli.sh [hunk 7]
```diff
@@ -0,0 +1,23 @@
+#!/bin/bash -eE
```
Comment (artemry-nv) [inline .ci/build_cli.sh:1]: You may need also `set -o pipefail` for robustness

File: .ci/configure_sharp.sh [hunk 8]
```diff
@@ -157,7 +159,9 @@ verify_sharp() {
         --map-by node \
         -x LD_LIBRARY_PATH \
         --allow-run-as-root \
+	-mca oob_tcp_if_exclude eth0 \
     "
+#	-x UCX_TLS=^sm \
```
Comment (artemry-nv) [inline .ci/configure_sharp.sh:164]: Remove commented code

File: .ci/configure_sharp.sh [hunk 9]
```diff
@@ -193,16 +196,17 @@ verify_sharp() {
         echo "FAIL"
         exit 1
     fi
+echo "::endgroup::"
```
Comment (artemry-nv) [inline .ci/configure_sharp.sh:199]: Move GH specific formatters into a variable and use it here (e.g. maybe useful if we'll use another CI framework)

File: .ci/ibdev2netdev [hunk 10]
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
```
Comment (artemry-nv) [inline .ci/ibdev2netdev:1]: Fix code formatting

Comment (artemry-nv) [inline .ci/ibdev2netdev:1]: Use `/bin/sh -eE` where it's possible or `/bin/bash -eE` if you use bash specific constructions

File: .ci/nccl_tests [hunk 11]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
```
Comment (artemry-nv) [inline .ci/nccl_tests:1]: Add comments for all the helper/wrapper scripts what they're for.

File: .github/workflows/nccl-sharp-plugin.yml [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+          - host01
+          - host02
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
+jobs:
+  build:
+
+    runs-on: [self-hosted, linux , x64]
```
Comment (artemry-nv) [inline .github/workflows/nccl-sharp-plugin.yml:21]: Odd space

**Review by artemry-nv — CHANGES_REQUESTED** [review]

**Comment by artemry-nv** [comment]
@B-a-S please squash into one commit

File: .ci/nccl_tests (hunk 11 above)
[the remark this answers is not in the document]
Comment (B-a-S) [reply .ci/nccl_tests:1]: These scripts use as wrappers, to fix issues with legacy

File: .ci/run_nccl_tests.sh [hunk 13]
```diff
@@ -51,6 +46,7 @@ MPIRUN_OPTIONS_COMMON="\
 -np $NP \
 --report-bindings \
 --allow-run-as-root \
+-mca oob_tcp_if_exclude eth0 \
```
[the remark this answers is not in the document]
Comment (B-a-S) [reply .ci/run_nccl_tests.sh:53]: This construction used in the old code

File: .github/workflows/nccl-sharp-plugin.yml [hunk 14]
```diff
@@ -0,0 +1,33 @@
+name: NCCL Sharp plugin CI
+
+on:
+  workflow_dispatch: 
+    inputs:
+      mainhost:
+        description: 'Choose one of hosts to run:'
+        required: true
+        type: choice
+        default: 'host01'
```
[the remark this answers is not in the document]
Comment (B-a-S) [reply .github/workflows/nccl-sharp-plugin.yml:10]: It's possible to choose which one of the hosts is running first.

File: .ci/build_cli.sh (hunk 7 above)
Comment (artemry-nv) [reply .ci/build_cli.sh:1]: Please add clarification for each remarks

File: .ci/build_cli.sh [hunk 15]
```diff
@@ -0,0 +1,22 @@
...
+. /GIT/pushd_functions.sh
+. /GIT/ci_functions.sh
+pushd /GIT
+case $1 in
+    build)
+	configure
```
Comment (artemry-nv) [inline .ci/build_cli.sh:7]: Please avoid tab symbols

File: .ci/build_cli.sh [hunk 16]
```diff
@@ -0,0 +1,22 @@
...
+        echo "Building NCCL sharp plugin"
```
Comment (artemry-nv) [inline .ci/build_cli.sh:8]: Fix formatting (e.g. you can use auto-formatting from PyCharm)

File: .ci/build_nccl_rdma_sharp_plugins.sh [hunk 17]
```diff
@@ -1,6 +1,9 @@
 #!/bin/bash -leE
```
Comment (artemry-nv) [inline .ci/build_nccl_rdma_sharp_plugins.sh:1]: Why do we need `-l` here?

File: .ci/ci_functions.sh [hunk 18]
```diff
@@ -0,0 +1,43 @@
+#!/bin/bash -eE
+# Preparation a workplace & configs to CI
+function configure() {
+    rm -rf "${NFS_WORKSPACE}" ||:
```
Comment (artemry-nv) [inline .ci/ci_functions.sh:4]: Maybe change to more popular w/a with `|| true`?

File: .ci/run_nccl_tests.sh [hunk 19]
```diff
@@ -1,45 +1,44 @@
 #!/bin/bash -leE
-
-SCRIPT_DIR="$( cd "$(dirname "$0")" ; pwd -P )"
+GH=${GH:-0}
+if [ "${GH}" -eq 1 ]; then
+    GH_FOLD="::group::"
```
Comment (artemry-nv) [inline .ci/run_nccl_tests.sh:4]: Please add a comment that this's GH Actions specific formatting macros

File: .ci/sharp_coll_test_wrapper [hunk 20]
```diff
@@ -0,0 +1,4 @@
+#!/bin/bash -eE
+# Wrapper to add correct parameter to the main scripts without refactoring
+TEST_DEV=$(ibdev2netdev | awk '{ print $1 }' ):1
+./sharp_coll_test -d "${TEST_DEV}" "${@}"
```
Comment (artemry-nv) [inline .ci/sharp_coll_test_wrapper:4]: Is it safe to use such relative paths here?

File: .ci/build_cli.sh (hunk 16 above)
Comment (B-a-S) [reply .ci/build_cli.sh:8]: Done

File: .ci/build_nccl_rdma_sharp_plugins.sh (hunk 17 above)
Comment (B-a-S) [reply .ci/build_nccl_rdma_sharp_plugins.sh:1]: It's legacy

File: .ci/build_cli.sh (hunk 7 above)
Comment (B-a-S) [reply .ci/build_cli.sh:1]: Pipe was removed

File: .ci/ci_functions.sh (hunk 1 above)
Comment (B-a-S) [reply .ci/ci_functions.sh]: Done

File: .ci/ci_functions.sh (hunk 2 above)
Comment (B-a-S) [reply .ci/ci_functions.sh]: Done

File: .ci/ci_functions.sh (hunk 3 above)
Comment (B-a-S) [reply .ci/ci_functions.sh]: Code was changed

File: .ci/ci_functions.sh (hunk 4 above)
Comment (B-a-S) [reply .ci/ci_functions.sh]: set 3

File: .ci/ci_functions.sh (hunk 5 above)
Comment (B-a-S) [reply .ci/ci_functions.sh]: Done

File: .ci/ci_functions.sh (hunk 6 above)
Comment (B-a-S) [reply .ci/ci_functions.sh]: Done

File: .ci/ibdev2netdev (hunk 10 above)
Comment (B-a-S) [reply .ci/ibdev2netdev]: Done

Comment (B-a-S) [reply .ci/ibdev2netdev]: Done

File: .ci/nccl_tests [hunk 21]
```diff
@@ -0,0 +1,9 @@
...
+IB_DEV=$(ibdev2netdev | awk '{ print $1 }' ):1
+ETH_DEV=$(ibdev2netdev | awk '{ print $5 }' )
+
+export HCOLL_MAIN_IB=${IB_DEV} 
+export NCCL_IB_HCA=${IB_DEV} 
+export UCX_NET_DEVICES=${IB_DEV} 
+export NCCL_SOCKET_IFNAME=${ETH_DEV}
+exec $@
```
[the remark this answers is not in the document]
Comment (B-a-S) [reply .ci/nccl_tests]: Fixed

File: .github/workflows/nccl-sharp-plugin.yml (hunk 12 above)
Comment (B-a-S) [reply .github/workflows/nccl-sharp-plugin.yml:20]: Done

File: .ci/build_cli.sh (hunk 15 above)
Comment (B-a-S) [reply .ci/build_cli.sh]: Fixed

File: .ci/run_nccl_tests.sh (hunk 19 above)
Comment (B-a-S) [reply .ci/run_nccl_tests.sh]: Added

File: .ci/sharp_coll_test_wrapper (hunk 20 above)
Comment (B-a-S) [reply .ci/sharp_coll_test_wrapper:5]: It's ok. The script is copied into dir with tests

File: .ci/ci_functions.sh (hunk 18 above)
Comment (artemry-nv) [reply .ci/ci_functions.sh:5]: Don't see the changes

File: .github/workflows/nccl-sharp-plugin.yml (hunk 12 above)
Comment (artemry-nv) [reply .github/workflows/nccl-sharp-plugin.yml:19]: Not fixed

Comment (artemry-nv) [reply .github/workflows/nccl-sharp-plugin.yml:19]: Check other similar places

**Review by artemry-nv — CHANGES_REQUESTED** [review]

File: .ci/ci_functions.sh (hunk 18 above)
Comment (B-a-S) [reply .ci/ci_functions.sh]: Sorry, done

**Review by artemry-nv — APPROVED** [review]

**Pull request merged**